dll/appcompat/apphelp/shimeng.c

```diff
@@ -597,9 +597,9 @@ FARPROC WINAPI StubGetProcAddress(HINSTANCE hModule, LPCSTR lpProcName)
     PHOOKMODULEINFO HookModuleInfo;
     FARPROC proc = ((GETPROCADDRESSPROC)g_IntHookEx[0].OriginalFunction)(hModule, lpProcName);
 
-    if (!HIWORD(lpProcName))
+    if ((DWORD_PTR)lpProcName <= MAXUSHORT)
```

Comment:
`HIWORD` is not correct here?

Reply:
HIWORD ignores the upper 32 bit on 64 bit systems. Alternatively IS_INTRESOURCE macro could be used, which is technically correct, but doesn't fit semantically and I couldn't find an appropriate macro.